docs/contributor/arch/016-transform-filter-support-for-telemetry-pipelines.md

```diff
@@ -0,0 +1,118 @@
+- `slient` : The processor ignores errors returned by conditions, does not log them, and continues on to the next condition.
+- `propagate` : The processor returns the error up the pipeline. This will result in the payload being dropped from the collector.
+
+The recommended error mode is `ignore` and this will be used as default configuration.
+
+The OTTL context will be embedded in the OTTL statements (in progress with [issue #29017](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29017)) and will be available in the upcoming beta version. The solution will not implement the context as configuration parameter.
+
+To ensure data consistency and sampling efficiency, the custom OTTL transformation and filtering processor will be near the end of the pipeline chain, before the exporters.
+
+See the OTel configuration:
+
+```yaml
+service:
+    pipelines:
+        metrics/test-output:
+            receivers:
+                - routing/test
+                - forward/test
+            processors:
+                - filter/drop-if-input-source-runtime
+                - filter/drop-if-input-source-prometheus
+                - filter/drop-if-input-source-istio
+                - transform/set-instrumentation-scope-kyma
+                - resource/insert-cluster-name
+                - resource/delete-skip-enrichment-attribute
+                - transform/custom-transform-processor
+                - filter/custom-filter-processor
+                - batch
+            exporters:
+                - otlp/test
+```
+
+### Risk
+
+The OTTL library is still in its alpha phase, meaning that updates to the package may introduce breaking changes, including syntax and semantic updates or bugs. This poses a risk of disrupting running pipelines.
+
+### Mitigation
+
+To mitigate this risk, we suggest the following measures:
+- The [beta version](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28892) of the OTTL library will be used in the OTel Collector to ensure that the library is stable and reliable.
+- New Versions of OTTL with Breaking Changes: The breaking changes of the OTTL library are released with backward compatibility with following the process described [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/coding-guidelines.md#breaking-changes), the backward compatibility provided via `feature gates`, those `feature gates` are available over several releases. This helps us adapt the changes and reduce the chance of widespread disruption. There is an exception breaking changes with public Go API, the breaking changes with public Go API most likely will not follow the process described [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/coding-guidelines.md#breaking-changes). Since those changes are not end-user-faced changes and may require changes in our implementation uses the public API. 
```

Comment:
```suggestion
- New Versions of OTTL with Breaking Changes: The breaking changes of the OTTL library are released with backward compatibility following the [breaking changes](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/coding-guidelines.md#breaking-changes) process. The backward compatibility is provided via `feature gates`. They are available in several releases. This helps us adapt to the changes and reduce the chance of widespread disruption. There are exceptional breaking changes with public Go API. These breaking changes most likely will not follow the breaking changes process. Since those changes are not end-user-faced, they may require changes in our implementation using the public API. 
```